### ISSUE TITLE
Can mapApp() have a progress indicator at startup for the download?

### ISSUE THREAD

**Issue by richardsc** [ArgoCanada/argoFloats#403]
Just wondering, sometimes it feels like an ETERNITY when I start up the app and wait for a map to show up -- long enough that I wonder if the whole thing has just hung.

Is there some kind of Shiny thing that could be used, with say the index file size, to give a progress bar for the download?

**Comment by dankelley** [comment]
I think J ought to look into that.  With the state of shiny 2 years ago (when I wrote the glider-qc-app), the shiny progress indicator was itself slow.  But maybe that is fixed now, but it was not tied to fraction-of-work-done, but rather it was just a sort of "beachball" thing.  Now, though, I think there are better ways to get progress bars in shiny.

**Comment by paleolimbot** [comment]
Just a quick note that you might be able to do this with `curl::curl_fetch_stream()` (I also have this problem when trying to get this to work from the DFO VPN) (ok ok...back to vacation!)

**Comment by richardsc** [comment]
Another option, though not as generally user friendly and not something that helps if it ever gets deployed on a server (hmmmm ... I just remembered that I have a Shiny server ... and could put in a cron job to just keep the index fresh ....) could be to spit out a progress bar to stdout.  That way if you run the app from a terminal (or in RSTudio) you can at least see what it's doing in the background.

**Comment by paleolimbot** [comment]
An easy fix to this might be to use `quiet = FALSE` when calling `getIndex()` from the Shiny app - I removed the progress bar in an earlier PR because it caused some very weird issues on Windows when cancelling the download, but it will at least give you a message  that tells you what file is being downloaded in the console.

**Comment by dankelley** [comment]
I think the good option might be to check on the age of the index file, and put up a dialog box saying something like *"The local Argo index file is 2 weeks old.  Do you want to download a new version, which will take a minute or two?"*. That way, if there's a cron job to download it every night at 3AM or something, all will be fine for the user who wants to stay up-to-date.  And, of course, some users will specifically *not* want a new version.

Also, I've noticed that when it's slow, my machine is not actually using much CPU or bandwidth.  Rather than giving data slowly, it seems to be just queuing the job for later.  This is just an impression though, and I don't know what to do about it.

**Comment by paleolimbot** [comment]
In argodata I have `argo_update_global()` and set the default `age` to `Inf` so that the user is always in charge of when to update the index. I found that this was a significant quality-of-life improvement for argodata's only user (me!).

**Comment by j-harbin** [comment]
Hi Team,

I did some work on this in commit db36f991b8c295792831a6f07d796b9962040d36 of the develop branch. I initially was working with `withProgress()` within `shiny`, but I personally found it was slowing things down. In saying that, I agree with Clark that it seemed like an eternity for everything to load up, and some times I didn't know it it was even loading. To fix this, I made messages come up (tweaks to messages Dan already had) which say:

1. Step 1/5: Getting "core" Argo index, either by downloading new data or using data in "destdir".  This may take a minute or two."
2. Step 2/5: Getting "BGC" Argo index, either by downloading new data or using cached data.  This may take a minute or two."
3. Step 3/5: Getting "Deep" Argo index. This may take a minute or two.
4. Step 4/5: Creating widgets
5. Step 5/5: Creating plot

This at least tells the user there are 5 steps they must get through, and they know what's going on. I'll leave this open for now, and Clark if you are happy with this, feel free to close it!
